topcoffea/modules/combine_json_batch.py

```diff
@@ -0,0 +1,36 @@
+import json
+import re
+
+'''
+Combine event information from multiple batches
+Sums `nEvents`, `nGenEvents`, and `nSumOfWeights`
+Does not update the list of files
+'''
+def combine_json_batch(batch_name):
+    if '_b' not in batch_name:
+        raise Exception(f"The file {batch_name} does not contain '_b' in the filename!")
+
+    jsonFile = open(batch_name,'r')
+    batch = json.load(jsonFile) # Load the JSON information for the batch file
+
+    nom_name = batch_name.replace('_b1','')
+    nom_name = re.sub(r'_b\d*', '_b1', batch_name) # '_b[0-9]' to '_b1' in the file name
+    try:
+        jsonFile = open(nom_name,'r')
+    except FileNotFoundError:
+        print(f'{nom_name} not found!')
+        return
+    nom = json.load(jsonFile) # Load the JSON information for the nominal file
+
+    for l in batch['files']:
+        nom['files'].append(l)
+
+    ''' Load the event counts, sum of weights, etc. '''
+    nom['nEvents']         = nom['nEvents']       + batch['nEvents']
+    nom['nGenEvents']      = nom['nGenEvents']    + batch['nGenEvents']
+    nom['nSumOfWeights']   = nom['nSumOfWeights'] + batch['nSumOfWeights']
```

Comment:
This looks right to me, but for some reason the JSONs modified by this PR seem to have the same number for `nSumOfWeights` and `nGenEvents` (and it seems to correspond to `nGenEvents`). Not sure how that's happening if they were generated with this code though.

Reply:
That's a good catch. I'm not sure why either, but I'll look into it.

Reply:
I was going through the blame in the history of these files, and it looks like the last update was when @kmohrman used `update_json_sow.py`. I don't think this works anymore, since the script seemed to use the `SumOfWeights` histogram we've removed from the processor.

Reply:
Hmm, so I think the `update_json_sow.py` stuff was written by @Andrew42 specifically to calculate the sum of the weights and put them in the JSON files, which is actually what allowed us to be able to drop the `SumOfWeights` hist (and that entire method of normalization) from the main processor to begin with. So, I'm not sure if the fact that the last update to the JSONs being the update to the sum of the weights could explain this particular issue.

Also, I think the `nSumOfWeights` listed in the JSONs prior to the merge is actually correct. I think it seems to be when the JSONs were merged together that the sum of the `nSumOfWeights` was apparently replaced by the sum of the `nGenEvents`. However, I could certainly be missing something.

Reply:
I learned how to run the `sow` scripts, and added them to the README. The sub m sum of weights should now be correct.